### PR TITLE
feat(api): Add "is" queries to the event/group search grammar

### PR DESCRIPTION
**Pull request by wedamija**
"is" queries are the main thing that our search grammar doesn't currently support, so adding it
as part of work on APP-961.

I didn't put any defensive code in `get_snuba_query_args` to ignore these params. Not sure it's
necessary, since we'll just treat them as tags if they get passed in an event query, which will
return no results.

### CODE REVIEW

File: src/sentry/api/event_search.py [hunk 1]
```diff
@@ -129,6 +132,16 @@ def is_wildcard(self):
 
 class SearchVisitor(NodeVisitor):
 
+    @cached_property
+    def is_filter_translators(self):
+        is_filter_translators = {
+            'assigned': (SearchKey('unassigned'), SearchValue(False)),
+            'unassigned': (SearchKey('unassigned'), SearchValue(True)),
```
Comment (wedamija) [inline src/sentry/api/event_search.py]: So far we haven't used `True/False` values in `SearchValue`. I think it makes sense here, open to changing it if anyone disagrees.

Comment (macqueen) [reply src/sentry/api/event_search.py]: makes sense to me, what would the alternative be?

**Review by macqueen — APPROVED** [review]
re: treating `is`, etc as tags in event search -- that seems fine to me for now at least. might be worth filing a ticket to follow up so we can decide if we want the search to be more aggressive about erroring when we think a user might be confused.

File: src/sentry/api/event_search.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/api/event_search.py]: No idea, I just noticed that so far we've only used strings, wasn't sure if there was a reason for it.

**Pull request merged**